snapshot/lib/snapshot/runner.rb

```diff
@@ -204,6 +208,20 @@ def erase_simulator(device_type)
       `xcrun simctl erase #{device_udid} &> /dev/null`
     end
 
+    def localize_simulator(device_type, language, locale)
+      device_udid = TestCommandGenerator.device_udid(device_type)
```

Comment:
This is not a problem specific to my changes. Most of `snapshot` assumes that `device_udid` is available and that `device_type` exists. See for example `erase_simulator` and `uninstall_app`.

My feeling is that if `device_udid` can be `nil`, it should not be checked in the individual functions but instead at a higher level.

I'm happy to improve the general robustness of this tool but I am not sure if that should be part of this PR? Let me know.